src/utils/errors/error-base.ts

```diff
@@ -1,13 +1,23 @@
 export abstract class ErrorBase extends Error {
   protected code: string;
+  protected details: { [key: string]: any };
 
-  public constructor(code: string, message: string) {
+  public constructor(
+    code: string,
+    details: { [key: string]: any },
```

Comment:
I would put details after message, and make it optional.

Reply:
That would also makes descendants cleaner
`
export class ValidationError extends ErrorBasePublic {
  constructor(message: string, details: { [key: string]: any }) {
    super(ErrorCodes.VALIDATION_ERROR, message, details);
  }
}
`